app/models/sensor.rb

```diff
@@ -2,6 +2,13 @@ class Sensor < ActiveRecord::Base
   has_many :readings, dependent: :destroy
   has_many :measurement_sensors, dependent: :destroy
   has_many :measurements, through: :measurement_sensors
+  has_many :locations, dependent: :destroy
 
-  validates :long, :lat, presence: true
+  before_create :generate_token
+
+  private
+
+  def generate_token
+    self.token = SecureRandom.urlsafe_base64.gsub('-_','')[0,12]
```

Comment:
If we are planning to migrate into RoR 5 in the feature than we can use this backport https://coderwall.com/p/kb97gg/secure-tokens-from-rails-5-to-rails-4-x-and-3-x (secure tokens will be available out of the box in 5).